### ISSUE TITLE
linter: preserve-caught-error false positive for AggregateError options argument

### ISSUE THREAD

**Issue by boundedagency** [oxc-project/oxc#25761]
### What version of Oxlint are you using?

1.78.0

### What command did you run?

```shell
oxlint -A all -D preserve-caught-error repro.js
```

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

No config file is required. The command above enables only `preserve-caught-error`.

### What happened?

Oxlint reports `eslint(preserve-caught-error)` for a valid built-in `AggregateError` whose `cause` is present in its third argument:

```js
async function repro(run, cleanup) {
  try {
    return await run();
  } catch (error) {
    try {
      await cleanup();
    } catch (cleanupError) {
      throw new AggregateError(
        [error, cleanupError],
        "cleanup failed",
        { cause: cleanupError },
      );
    }
  }
}
```

Actual diagnostic:

```text
error eslint(preserve-caught-error): There is no cause error attached to this new thrown error.
```

Expected: no diagnostic. `AggregateError` has the signature `AggregateError(errors, message, options)`, so the cause options object is argument index 2. ESLint 10.8.0 accepts the reproduction above.

A minimal control matrix isolates the behaviour:

```js
try {
  doSomething();
} catch (error) {
  // Oxlint 1.78.0 reports; ESLint 10.8.0 accepts.
  throw new AggregateError([error], "aggregate", { cause: error });
}

try {
  doSomething();
} catch (error) {
  // Both accept.
  throw new Error("wrapped", { cause: error });
}

try {
  doSomething();
} catch (error) {
  // Both correctly report.
  throw new AggregateError([error], "aggregate");
}
```

The rule currently checks `args.get(1)` for the options object for all supported constructors:

https://github.com/oxc-project/oxc/blob/oxlint_v1.78.0/crates/oxc_linter/src/rules/eslint/preserve_caught_error.rs#L72-L76

The fixer in the same rule already recognises that `AggregateError` takes options as its third argument:

https://github.com/oxc-project/oxc/blob/oxlint_v1.78.0/crates/oxc_linter/src/rules/eslint/preserve_caught_error.rs#L103-L143

ESLint selects options index 2 for the built-in `AggregateError`:

https://github.com/eslint/eslint/blob/v10.8.0/lib/rules/preserve-caught-error.js#L433-L455

The ECMAScript constructor signature is `AggregateError(errors, message [, options])`:

https://tc39.es/ecma262/2026/multipage/ordinary-and-exotic-objects-behaviours.html#sec-aggregate-error-constructor

The same `args.get(1)` lookup is still present on Oxc `main` at the time of filing.
